core/src/test/java/org/elasticsearch/cluster/ClusterChangedEventTests.java

```diff
@@ -115,29 +117,25 @@ public void testIndicesMetaDataChanges() {
         final int numIterations = 5;
         ClusterState previousState = createState(numNodesInCluster, randomBoolean(), initialIndices);
         for (int i = 0; i < numIterations; i++) {
-            final int numAdd = randomIntBetween(0, 5); // add random # of indices to the next cluster state
-            final List<Index> stateIndices = new ArrayList<>();
-            for (Iterator<IndexMetaData> iter = previousState.metaData().indices().valuesIt(); iter.hasNext();) {
-                stateIndices.add(iter.next().getIndex());
-            }
-            final int numDel = randomIntBetween(0, stateIndices.size() - 1); // delete random # of indices from the next cluster state
-            final List<Index> addedIndices = addIndices(numAdd, i);
-            List<Index> delIndices = delIndices(numDel, stateIndices);
-            final ClusterState newState = nextState(previousState, addedIndices, delIndices, 0);
-            ClusterChangedEvent event = new ClusterChangedEvent("_na_", newState, previousState);
-            final List<String> addsFromEvent = event.indicesCreated();
-            List<Index> delsFromEvent = event.indicesDeleted();
-            assertThat(new HashSet<>(addsFromEvent), equalTo(addedIndices.stream().map(Index::getName).collect(Collectors.toSet())));
-            assertThat(new HashSet<>(delsFromEvent), equalTo(new HashSet<>(delIndices)));
-            assertThat(event.metaDataChanged(), equalTo(addedIndices.size() > 0 || delIndices.size() > 0));
-            final IndexGraveyard newGraveyard = event.state().metaData().indexGraveyard();
-            final IndexGraveyard oldGraveyard = event.previousState().metaData().indexGraveyard();
-            assertThat(((IndexGraveyard.IndexGraveyardDiff)newGraveyard.diff(oldGraveyard)).getAdded().size(), equalTo(delIndices.size()));
+            final ClusterState newState = executeIndicesChangesTest(previousState, i, i + 1 == numIterations);
             previousState = newState; // serves as the base cluster state for the next iteration
         }
     }
 
     /**
+     * Test that the indices deleted list is correct when the previous cluster state is
+     * not initialized/recovered.  This should trigger the use of the index tombstones to
+     * determine the deleted indices.
+     */
+    public void testIndicesDeletionWithNotRecoveredState() {
+        final int numIterations = 5;
```

Comment:
The **@Repeat** annotation is forbidden by the precommit check.  I understood the argument for not needlessly having iterations in tests, but in this case, I thought it is useful, namely because I can check different addition/deletion list sizes in one check, including edge cases.  But I can change the test to be more explicit and cover all edge cases separately within the test.

The number of iterations in the `testIndicesMetaDataChanges` is more crucial because I want to use the previous state as a basis for the creation of the next state.